### PR TITLE
Feature/dfgintegration

### LINKED ISSUE #27: Fix lost marginal issue

### PR DESCRIPTION
**Pull request by GearsAD**
This refactors IncrementalInference for DistributedFactorGraphs. More to follow.

**Note** : Resolves #27

### CODE REVIEW

**Comment by dehann** [comment]
cc @tonioteran changes to Bayes tree are under way and these will be merged into master branch in the next couple of days.  The recently released IIF v0.6.x has has a new option in `batchSolve!(fg, treeinit=true)` that changes the tree based solution to use a state machine rather than messy for-loops of old.  

The upcoming IIF v0.7.x which is (heavily dependent on this PR) will change to use JuliaRobotics/DistributedFactorGraphs.jl as the primary graphing interface.  References to `Graphs.ExVertex` will be limited to only the Bayes tree, and all other `FactorGraph` related uses will be changed to DFG types and API instead.

**Comment by dehann** [comment]
current dev example:
https://github.com/JuliaRobotics/IncrementalInference.jl/blob/6570bd42b05c07f617b63170519892928ba7e503/test/unittests/FactorGraph01_tests.jl#L1

**Comment by dehann** [comment]
Alright @GearsAD , I'm able to run the example and will pick up on the final merge for tree and inference, starting with the current error on the last line of example (Ln61), with issue in function `getCliqPotentials!`.  Ball is in my court thanks.

FYI, I changed Manifest.toml to point DFG to /home/dehann rather than /home/GearsAD

**Comment by dehann** [comment]
There are still a few quirks:
- [ ] with unexpected error when using multi-process state machine based solving, and 
- [ ] a recently identified regression on the solve quality (see issue JuliaRobotics/Caesar.jl#335),
- [ ] plotting support with RoMEPlotting v0.2.x not ready yet.

IIF v0.7.x will switch to clique state machine only and deprecate previous stack for loop when solving on tree entirely.  Need to merge this branch soon however, so that unified progress can be made on new v0.7.x-alpha version of code.

> TLDR; this PR will break master branch functionality in a few places

**Comment by GearsAD** [comment]
:+1:
